pandas/_libs/tslib.pyx

```diff
@@ -1577,7 +1589,17 @@ cpdef convert_str_to_tsobject(object ts, object tz, object unit,
                     return obj
                 else:
                     # Keep the converter same as PyDateTime's
-                    ts = Timestamp(obj.value, tz=obj.tzinfo)
+                    obj2 = convert_to_tsobject(obj.value, obj.tzinfo,
+                                               None, 0, 0)
+                    dtime = datetime(obj2.dts.year, obj2.dts.month,
+                                     obj2.dts.day,
+                                     obj2.dts.hour, obj2.dts.min, obj2.dts.sec,
+                                     obj2.dts.us, obj2.tzinfo)
+                    obj2 = _convert_datetime_to_tsobject(dtime, tz)
```

Comment:
have ``convert_datetime_to_tsobject`` take an optional ``ps=`` to facilitate this type of post-conversion.

also then you don't need a ``obj2`` avoiding more confusion here.

Reply:
Good idea.